e2e/cypress/integration/files_and_attachments/upload_files_spec.js

```diff
@@ -202,6 +205,79 @@ describe('Upload Files', () => {
+                EnablePublicLink: true,
+            },
+        });
+
+        const attachmentFilename = 'image-file.jpg';
+
+        // # Make a post with a file attached
+        cy.get('#fileUploadInput').attachFile(attachmentFilename);
+        cy.postMessage('Post with attachment to be deleted');
+
+        // # Get the last post
+        cy.getLastPostId().then((lastPostId) => {
+            // # Scan inside of the last post message
+            cy.get(`#${lastPostId}_message`).and('be.visible').within(() => {
+                // * Check if the attached image is in the post and then click to open the preview
+                cy.findByLabelText(`file thumbnail ${attachmentFilename}`).should('be.visible').click();
+            });
+        });
+
+        // * Verify preview modal is opened
+        cy.get('.a11y__modal').should('exist').and('be.visible').
+            within(() => {
+            // * Check if get public link button is present and click it
+                cy.findByText('Get a public link').should('be.visible').click({force: true});
+            });
+
+        cy.wait(TIMEOUTS.ONE_SEC);
+
+        // * Verify copy public link modal is opened
+        cy.get('.a11y__modal.modal-dialog').should('exist').and('be.visible').
+            within(() => {
+                // * Verify that copy link button is present
+                cy.findByText('Copy Link').should('be.visible');
+
+                // # Get the copy link of the attachment and save for later purpose
+                cy.get('#linkModalTextArea').invoke('text').as('publicLinkOfAttachment');
+            });
+
+        // # Close the image preview modal
+        cy.get('body').type('{esc}');
+
+        // # Once again get the last post with attachment, this time to delete it
+        cy.getLastPostId().then((lastPostId) => {
+            // # Click post dot menu in center.
+            cy.clickPostDotMenu(lastPostId);
+
+            // # Scan inside the post menu dropdown
+            cy.get(`#CENTER_dropdown_${lastPostId}`).should('exist').within(() => {
+                // # Click on the delete post button from the dropdown
+                cy.findByText('Delete').should('exist').click();
+            });
+        });
+
+        // * Verify caution dialog for delete post is visible
+        cy.get('.a11y__modal.modal-dialog').should('exist').and('be.visible').
+            within(() => {
+                // # Confirm click on the delete button for the post
+                cy.findByText('Delete').should('be.visible').click();
+            });
+
+        // # Try to fetch the url of the attachment we previously deleted
+        cy.get('@publicLinkOfAttachment').then((publicLinkOfAttachment) => {
+            cy.request({url: publicLinkOfAttachment, failOnStatusCode: false}).then((response) => {
+                // * Verify that the link no longer exists in the system
+                expect(response.status).to.be.equal(404);
+            });
```

Comment:
I think this is a bonus but it's missing the UI verification on "You should see an error displayed "Error, Unable to get the file info, Back to Mattermost".